test/sql/types/nested/map/map_entries/basic.test

```diff
@@ -0,0 +1,53 @@
+# name: test/sql/types/nested/map/map_entries/basic.test
+# description: Show expected behavior with an empty arguments list
+# group: [map_entries]
```

Comment:
Can you add tests that use the functions in subqueries/filters? Also, can you add tests that take dictionary vector as their input? I would also like to see some more different types, and more deeply nested types. Can you also try to use this in a lambda function? And in a macro?